plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -251,38 +249,39 @@ public MetadataEntry getMetadataEntry(TableSnapshot tableSnapshot, ConnectorSess
     public List<AddFileEntry> getActiveFiles(TableSnapshot tableSnapshot, ConnectorSession session)
     {
         try {
-            TableLocation tableLocation = new TableLocation(tableSnapshot.getTable(), tableSnapshot.getTableLocation());
-            DeltaLakeDataFileCacheEntry cachedTable = activeDataFileCache.get(tableLocation, () -> {
+            TableVersion tableVersion = new TableVersion(new TableLocation(tableSnapshot.getTable(), tableSnapshot.getTableLocation()), tableSnapshot.getVersion());
+
+            DeltaLakeDataFileCacheEntry cacheEntry = activeDataFileCache.get(tableVersion, () -> {
```

Comment:
I just tried out a simple  scenario

```
trino> create table delta.tiny.deltatable1(c integer);
CREATE TABLE
trino> insert into delta.tiny.deltatable1 values 1;
INSERT: 1 row

trino> select * from delta.tiny.deltatable1;
 c 
---
 1 

trino> insert into delta.tiny.deltatable1 values 2;
INSERT: 1 row

trino> select * from delta.tiny.deltatable1;
 c 
---
 1 
 2 

trino> insert into delta.tiny.deltatable1 values 3;
INSERT: 1 row

trino> select * from delta.tiny.deltatable1;
 c 
---
 1 
 2 
 3
```

The `activeDataFileCache` had `3` token keys  corresponding to `delta.tiny.deltatable1`.
This doesn't seem right. Why do we cache the active data files for all the table versions?

Reply:
that's how caches work -- unless there is explicit eviction, they can contain garbage.

in fact, why would i proactively evict old entries? they can be used for queries that started earlier (reading older table version), or by time-travel queries (once we support them -- now we're one step closer to that)